src/main/drivers/dshot.c

```diff
@@ -139,6 +141,33 @@ uint16_t getDshotTelemetry(uint8_t index)
     return dshotTelemetryState.motorState[index].telemetryValue;
 }
 
+uint16_t erpmToRpm(int erpm)
+{
+    //  rpm = (erpm * 100) / (motorConfig()->motorPoleCount / 2)
+    return (erpm * 200) / motorConfig()->motorPoleCount;
+}
+
+// Caller shall ensure that dshot telemetry is active before calling this function (i.e. erpm data has already received)
+uint16_t getDshotAverageRpm(void)
+{
+    const int motor_count = getMotorCount();
+    static uint16_t averageRpm = 0;
+
+    // Calculate combined data only when necessary
+    if (dshotTelemetryState.combinedDataNeedsUpdate)
+    {
+        averageRpm = 0;
+        for (int i = 0; i < motor_count; i++) {
+            averageRpm += dshotTelemetryState.motorState[i].telemetryValue;
+        }
+        averageRpm = erpmToRpm(averageRpm / motor_count);
+
+        dshotTelemetryState.combinedDataNeedsUpdate = false;
+    }
+
+    return averageRpm;
+}
+
```

Comment:
```suggestion
static uint16_t averageRpm = 0;

// Caller shall ensure that dshot telemetry is active before calling this function (i.e. erpm data has already received)
void updateDshotAverageRpm(void)
{
    const int motor_count = getMotorCount();
    averageRpm = 0;
    for (int i = 0; i < motor_count; i++) {
        averageRpm += dshotTelemetryState.motorState[i].telemetryValue;
    }
    averageRpm = erpmToRpm(averageRpm / motor_count);
}

uint16_t getDshotAverageRpm(void)
{
    return averageRpm;
}

```

Reply:
@KarateBrot  In my side this change it is taking almost 1kb of additional flash. Please, could you check at your side?

```
Linking STM32F405 
Memory region         Used Size  Region Size  %age Used
           FLASH:         392 B        10 KB      3.83%
FLASH_CUSTOM_DEFAULTS:           8 B         6 KB      0.13%
    FLASH_CONFIG:          0 GB        16 KB      0.00%
          FLASH1:      557507 B       976 KB     55.78%
FLASH_CUSTOM_DEFAULTS_EXTENDED:          0 GB        16 KB      0.00%
   SYSTEM_MEMORY:          0 GB        29 KB      0.00%
             RAM:      106524 B       128 KB     81.27%
             CCM:       17620 B        64 KB     26.89%
     BACKUP_SRAM:          0 GB         4 KB      0.00%
       MEMORY_B1:          0 GB         0 GB
   text	   data	    bss	    dec	    hex	filename
 550527	   7380	 116376	 674283	  a49eb	./obj/main/betaflight_STM32F405.elf
Creating HEX ./obj/betaflight_4.4.0_STM32F405.hex 
make[2]: se sale del directorio '/home/blue/proyectos/betaflight'
make[1]: se sale del directorio '/home/blue/proyectos/betaflight'
Building STM32F405 succeeded.
```

Reply:
### On F7X2:
```
AXIM_FLASH1: -64 B
DTCM_RAM:     -4 B
```
`AXIM_FLASH1` is tight on the F7X2 so 64B of saved is a plus.

### On F405:
```
FLASH1: +52 B
RAM:     -4 B
CCM:     -4 B
```
We have more than enough `FLASH1` on F405 so the small increase is fine.

### On F411:
```
FLASH1: -56 B
```
`FLASH1` is tight on the F411 so 56B saved is a plus.

It seems like the compiler is aware when it needs to optimize for flash space or when it doesn't.

Reply:
Reworked.

Reply:
Is there a reason why you don't want to update the `averageRpm` in `dshot.c` and why you want to hide the implementation in `dshot_bitbang.c` and `pwm_output_dshot_shared.c`? I guess you don't want people to be able to update `averageRpm` outside of these two files?

Reply:
Yes, basically it was the reason, and because when reading more than 1 stored telemetry values inside updateAverageRpm was the reason why in my side image size increased by 1kb. So I decided to take advantage that on these files telemetry was already being updated and avoiding using dshot telemetry stat structure members twice to prevent flash size to grow.